qubesbuilder/component.py

```diff
@@ -29,9 +29,27 @@
 import yaml
 from packaging.version import Version, InvalidVersion
 
-from qubesbuilder.common import sanitize_line
+from qubesbuilder.common import sanitize_line, STAGES
 from qubesbuilder.exc import ComponentError
 
+FORBIDDEN_PATTERNS = [".."]
+for s in STAGES:
+    FORBIDDEN_PATTERNS += [f".{s}.yml", f".{s}.yaml"]
+
+
+def deep_check(data):
+    if isinstance(data, dict):
+        for k, v in data.items():
+            deep_check(k)
+            deep_check(v)
+    elif isinstance(data, list):
+        for l in data:
+            deep_check(l)
+    elif isinstance(data, str):
+        for p in FORBIDDEN_PATTERNS:
+            if p in data:
+                raise ValueError(f"Forbidden pattern '{p}' found in '{data}'.")
```

Comment:
I'd prefer this to handle all possible (expected) types, and fail if unexpected type is found. IIUC it requires adding:
```
elif isinstance(data, int):
    pass
else:
    raise ValueError(f"Unexpected data type {type(data)} found")
```